docs/development/workflow/development_workflow.rst

```diff
@@ -149,6 +153,45 @@ document:
   ``astropy``; in prior versions of this documentation it was referred to as
   ``upstream``.
 
+
+.. _pre-commit:
+
+Pre-commit
+**********
+
+All of the coding style checks described in :ref:`code-style` can be performed automatically
+when you make a git commit using our provided `pre-commit hook <https://pre-commit.com/>`_
+for git, for more information see
+`git hooks <https://git-scm.com/book/en/v2/Customizing-Git-Git-Hooks#_git_hooks>`_.
+We encourage you to setup and use these hooks to ensure that your code always meets
+our coding style standards. This can be done by installing ``pre-commit`` in the root
+of your astropy repository by running::
+
+    pip install pre-commit
+    pre-commit install
+
+For more detailed instructions on installing ``pre-commit``, see the
+`install guide <https://pre-commit.com/#install>`_. Once this installation is
+complete, all the coding style checks will be run each time you commit and the
+necessary changes will automatically be applied to your code if possible.
+
+.. note::
+  The changes made by ``pre-commit`` will not be automatically staged, so you
+  will need to review and re-stage any files that ``pre-commit`` has changed.
+
+In general, git will not allow you to commit until the ``pre-commit`` hook has
+run successfully. If you need to make a commit which fails the ``pre-commit`` checks,
+you can skip (strongly discouraged) these checks by running::
```

Comment:
I don't think we need to "strongly discourage" this. We don't require every commit to be clean, just the final one that gets merged. It might be convenient during development to just make an intermediate commit without running the checks.